### ISSUE TITLE
False positive in PERF402 when append is more complex. 

### ISSUE THREAD

**Issue by bgianfo** [astral-sh/ruff#5494]
The new rule [PERF402](https://beta.ruff.rs/docs/rules/manual-list-copy/) suggests to convert lists to trivial copies in cases where it's not really possible. Ideally the tool would not report such false positives. 

If you're filing a bug report, please consider including the following information:

* A minimal code snippet that reproduces the bug.

```python
 # In this example you can't trivially convert this to a `list.copy` as the rule suggests.  
 for r in record_dict["values"]:
        list_of_stuff[r["other_value"]].append(r)
```

* The command you invoked (e.g., `ruff /path/to/file.py --fix`), ideally including the `--isolated` flag.

	`ruff --select=PERF402 example.py` 

* The current Ruff settings (any relevant sections from your `pyproject.toml`).

 	`PERF` ruleset is enabled

* The current Ruff version (`ruff --version`).

 	version = `0.0.276`

**Comment by charliermarsh** [comment]
Makes sense, we should probably require that the left-hand side expression is a static name (or at least an expression that doesn't depend on the iteration key). Thanks!
